### ISSUE TITLE
Chromedriver testing with nw.gui not working

### ISSUE THREAD

**Issue by anjanapaulose** [nwjs/nw.js#3899]
Hi,

I setup protractor for testing as explained in https://github.com/nwjs/nw.js/issues/2229 (Chromedriver with Karma [$100 awarded]).  
I am able to test applications with node modules. But  when I use anything from "nw.gui", its failing.

Other node modules like fs, and my custom module which has methods to be invoked are working. When I try, var nw = require("nw.gui"), its  returns a json object like {.App,.., window,..}, The issue if I accessnext level, nw.App or anything, the result is a json object which has only strings , the inner contents are not json objects anymore. eg: I get nw.App , but not nw.App.manifest.
Please help

**Comment by reynoldsalec** [comment]
For @anjanapaulose or anyone else who is trying to run Protractor tests with NWJS 0.13+, with the change over to the chrome-extension:// protocol, I noticed I had to change a couple things:

- Even if you've set baseUrl in the config file, that baseUrl doesn't seem to automatically append to the paths you call using browser.get():

```
browser.get('/dashboard'); // v0.12
browser.get(browser.baseUrl + '#/dashboard'); // v0.13+
```

- I'm hoping there's a way around this, but for me the only successful baseUrl I could set was 'chrome-extension://noakblofbajciaghholgljpkieiennnn/index.html':

```
var baseUrl = 'file://'+ path.resolve('build/index.html') + '#'; // v0.12
var baseUrl = 'chrome-extension://noakblofbajciaghholgljpkieiennnn/index.html'; // v0.13+
```

Once I switched these over and made sure to replace any other references to the "file://" protocol, my tests ran as they did on v0.12.

**Comment by gtrevg** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  gtrevg@gmail.com
    Domain plentyproductions.com has exceeded the max emails per hour (200/200 (100%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext5.iad.github.net ([192.30.252.196]:34329 helo=github-smtp2a-ext-cp1-prd.iad.github.net)
	by server21.abstractdns.com with esmtps (TLSv1:DHE-RSA-AES256-SHA:256)
	(Exim 4.87)
	(envelope-from <noreply@github.com>)
	id 1bE6vH-0007Gt-6W
	for github@trevorgattis.com; Sat, 18 Jun 2016 03:27:51 +0000
Date: Thu, 16 Jun 2016 16:19:49 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1466119189;
	bh=r1rm7SXoTz04WRcW9jqIM2wxNturAJfF4tiupS5fCrs=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=AkKnNmUDVUP/tl6BD/+Gd45qpRpnINn59i+6M0q8T98WV98H/M1l4KiarRS1qs3Cv
	 VnikJAxYL7YVwBKzHUdcQMDmoFcXDMTLfcEoOqrjnulhP1n3NkGmoSnGFOmUv5fM56
	 pAv/TgRLzPHaT6Gr4AdwRbPiEgq4ewN+BvCrtgbg=

**Comment by anjanapaulose** [comment]
I was trying with nw v0.12.3 .  I was able to resolve the issue with latest chromedriver using the nwapp args to chromeOptions , as mentioned in #3167.  This is my protractor.conf which is working.

_exports.config = {
  chromeDriver: '<path to nw chromedriver.exe>', 
  directConnect: true,
   //Capabilities to be passed to the webdriver instance.
  capabilities: {
    'browserName': 'chrome',
    "chromeOptions": {
      // The "--" is automatically prepended
      args: ['nwapp=<path to nw app folder>'],
      // It took me too long to figure out where this line belonged in chromeOptions
      binary: '<path to nw exe>'
    }
  },

   specs: ['test/component/**/*.spec.js'],

  framework: 'jasmine',
  // Options to be passed to Jasmine-node.
  jasmineNodeOpts: {
    showColors: true,
    defaultTimeoutInterval: 30000
  },
  
  rootElement: 'body', // specify a correct element where you bootstrap your AngularJS app, 'body' by default
 
};_


I had to remove baseURL, and onPrepare block from configuration. If that is specified, protractor will try to load it as normal angular app from baseURL, which again fails to give nw object. 
This configuration will load the nw app using package.json specifed in the app folder, so everything works fine.
